<h1>Задание к уроку 3:</h1> <h3>"Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных"</h3>

__1) Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.__

Среднее арифметическое:

$$ M(X) = \frac{1}{n}\cdot\sum_{j=1}^{n} x_{i} = \frac{1}{n}\cdot(x_{1}+\cdots+x_{n}) = $$

$$ = \frac{(100 + 80 + 75 + 77 + 89 + 33 + 45 + 25 + 65 + 17 + 30 + 24 + 57 + 55 + 70 + 75 + 65 + 84 + 90 +150)}{20} = $$

$$ = \frac{1306}{20} = 65.3 $$

Cреднее квадратичное отклонение:

$$ S = \sqrt[]{\frac{\sum_{n}^{i=1}\cdot(x_{i}-\overline{x})^{2}}{n-1}} = \sqrt[]{\frac{19002.2}{20-1}} = \sqrt[]{1000.12} = 31.62 $$

Несмещенная оценка дисперсий:

$$ S^{2} = \frac{\sum_{n}^{i=1}\cdot(x_{i}-\overline{x})^{2}}{n-1} = \frac{19002.2}{20-1} = 1000.12$$

Смещенная оценка дисперсий:

$$ S^{2} = \frac{\sum_{n}^{i=1}\cdot(x_{i}-\overline{x})^2}{n} = \frac{19002.2}{20} = 950.11$$

In [30]:
import numpy as np
from math import sqrt
salary = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
salary.sort()
print(f'Количестко измерений выборки: {len(salary)} измерений')
print(f'Среднее арифметическое значение равно: {sum(salary)/len(salary)}')
print(f'Среднее квадратичное отклонение равно: {round(np.std(salary, ddof=1), 2)}')
print(f'Несмещенная оценка дисперсии для данной выборки: {round(np.var(salary, ddof=1), 2)}')
print(f'Смещенная оценка дисперсии для данной выборки: {round(np.var(salary), 2)}')

Количестко измерений выборки: 20 измерений
Среднее арифметическое значение равно: 65.3
Среднее квадратичное отклонение равно: 31.62
Несмещенная оценка дисперсии для данной выборки: 1000.12
Смещенная оценка дисперсии для данной выборки: 950.11


__2) В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?__

Вероятность, что 3 белых мяча будут вытащены из второй корзины:

$$ p1 = \frac{C_{3}^{2}}{C_{8}^{2}}\cdot\frac{C_{5}^{3} \cdot C_{7}^{1}}{C_{12}^{4}} = \frac{\frac{3!}{2!\cdot(3-2)!}}{\frac{8!}{2!\cdot(8-2)!}}\cdot\frac{\frac{5!}{3!\cdot(5-3)!} \cdot \frac{7!}{1!\cdot(7-1)!}}{\frac{12!}{4!\cdot(12-4)!}} = \frac{3}{28}\cdot\frac{10\cdot7}{495} = 0.0(15)$$

Вероятность, что будут вытащены 2 белых мяча из первой корзины и 1 из второй:

$$ p2 = \frac{C_{5}^{2}}{C_{8}^{2}}\cdot\frac{C_{5}^{1} \cdot C_{7}^{3}}{C_{12}^{4}} = \frac{\frac{5!}{2!\cdot(5-2)!}}{\frac{8!}{2!\cdot(8-2)!}}\cdot\frac{\frac{5!}{1!\cdot(5-1)!} \cdot \frac{7!}{3!\cdot(7-3)!}}{\frac{12!}{4!\cdot(12-4)!}} = \frac{10}{28}\cdot\frac{5\cdot35}{495} = 0.1(26) $$

Вероятность, что будут вытащены 1 белый мяч из первой корзины и 2 из второй:

$$ p3 = \frac{C_{3}^{1} \cdot C_{5}^{1}}{C_{8}^{2}}\cdot\frac{C_{5}^{2} \cdot C_{7}^{2}}{C_{12}^{4}} = \frac{\frac{3!}{1!\cdot(3-1)!} \cdot \frac{5!}{1!\cdot(5-1)!}}{\frac{8!}{2!\cdot(8-2)!}}\cdot\frac{\frac{5!}{2!\cdot(5-2)!} \cdot \frac{7!}{2!\cdot(7-2)!}}{\frac{12!}{4!\cdot(12-4)!}} = \frac{3\cdot5}{28} \cdot\frac{10\cdot21}{495} = 0.2(27)$$

Вероятность, что 3 из 4-х вытащенных мяча будут белыми равна:

$$ P = p1+p2+p3 = 0.3(68) $$

In [37]:
from math import factorial
def combinations(n, k):
    result = factorial(n)/(factorial(k)*factorial(n-k))
    return result

p1 = combinations(3, 2)/combinations(8, 2)*(combinations(5, 3)*combinations(7, 1)/combinations(12, 4)) # ни одного белого из первой корзины
p2 = (combinations(3, 1)*combinations(5, 1)/combinations(8, 2))*(combinations(5, 2)*combinations(7, 2)/combinations(12, 4))# один белый из первой корзины
p3 = combinations(5, 2)/combinations(8, 2)*(combinations(5, 1)*combinations(7, 3)/combinations(12, 4)) # два белых из первой корзины

print(f'Вероятность, что 3 белых мяча будут вытащены из второй корзины равна: {p1}')
print(f'Вероятность, что 2 белых мяча будут вытащены из первой корзины и 1 из второй равна: {p3}')
print(f'Вероятность, что 1 белый мяч будет вытащены из первой корзины и 2 извторой равна: {p2}')
print(f'Вероятность, что 3 из 4-х вытащенных мяча будут белыми равна: {p1+p2+p3}')

Вероятность, что 3 белых мяча будут вытащены из второй корзины равна: 0.01515151515151515
Вероятность, что 2 белых мяча будут вытащены из первой корзины и 1 из второй равна: 0.12626262626262627
Вероятность, что 1 белый мяч будет вытащены из первой корзины и 2 извторой равна: 0.22727272727272727
Вероятность, что 3 из 4-х вытащенных мяча будут белыми равна: 0.3686868686868687


__3) На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:__

a). первым спортсменом

$$ p = 0.9  $$

б). вторым спортсменом

в). третьим спортсменом.

__4) В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:__

 a). на факультете A

б). на факультете B 

в). на факультете C?

__5) Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:__

$$ p_{1} = 0.1; \,\, p_{2} = 0.2; \,\, p_{3} = 0.25$$

#### а). все детали 

$$ P = p_{1} \cdot p_{2} \cdot p_{3} = 0.1\cdot0.2\cdot0.25 = 0.005$$

#### б). только две детали 

Вероятности того, что станки не выйдут из строя равны:

$$ q_{1} = 1-p_{1} = 0.9; \,\, q_{2} = 1-p_{2} = 0.8; \,\, q_{3} = 1-q_{3} = 0.75\,\,$$

Тогда вероятность того, что выйдет из строя только две детали равна:

$$ P = q_{1} \cdot p_{2} \cdot p_{3} + p_{1} \cdot q_{2} \cdot p_{3} + p_{1} \cdot p_{2} \cdot q_{3} = $$

$$ 0.9\cdot0.2\cdot0.25 + 0.1\cdot0.8\cdot0.25 + 0.1\cdot0.2\cdot0.75 = 0.045 + 0.02 + 0.015 = 0.08 $$

#### в). хотя бы одна деталь 

Найдём вероятность, что за месяц не выйдет из строя ни одна деталь:

$$ \overline P = \overline P_{1} \cdot \overline P_{2} \cdot \overline P_{3} = (1 - 0.1)\cdot (1 - 0.2)\cdot (1 - 0.25) = 0.9 \cdot 0.8 \cdot 0.75 = 0.54$$

Тогда вероятность того, что хотябы один станок выйдет из строя равна:

$$ P = 1 - \overline P = 1 - 0.54 = 0.46 $$

#### г). от одной до двух деталей?

Вероятность выхода из строя строго от одной до двух деталей можно получить, вычтя из вероятности выхода из строя хотябы одной детали(включающей выход из строй 2-х и 3-х деталей) вероятность выхода из строя 3-х(то есть всех) деталей:

$$ P = 0.46-0.005 = 0.455 $$